plugins/randpa_plugin/include/eosio/randpa_plugin/round.hpp

```diff
@@ -155,33 +168,35 @@ class randpa_round {
         state = state_type::prevote;
 
         auto last_node = tree->get_last_inserted_block(primary);
-
         if (!last_node) {
             randpa_wlog("Not found last node in tree for primary, primary: ${p}", ("p", primary));
             return;
         }
-
         auto chain = tree->get_branch(last_node->block_id);
 
         auto prevote = prevote_type { num, chain.base_block, std::move(chain.blocks) };
-        auto msg = prevote_msg(prevote, signature_providers);
-        add_prevote(msg);
-        prevote_bcaster(msg);
+
+        for (const auto& sig_prov : signature_providers) {
+            auto msg = prevote_msg(prevote, { sig_prov });
+            add_prevote(msg);
+            prevote_bcaster(msg); //TODO: move out of the loop
```

Comment:
I will try to move it out of the loop to bcast a whole msg (with all signatures).

Reply:
yeap, it would really great if we bcast already "aggregated" message